runtime/lib/capi/RuntimeCAPI.cpp

```
@@ -1031,4 +1031,26 @@ int8_t *__catalyst__rt__array_get_element_ptr_1d(QirArray *ptr, int64_t idx)
     QubitIdType *data = qubit_vector_ptr->data();
     return (int8_t *)&data[idx];
 }
+
+// -------------------------------------------------------------------------- //
+// MBQC Runtime CAPI
+// -------------------------------------------------------------------------- //
+
+// NOTE: Currently this runtime operations is exactly the same as __catalyst__qis__Measure();
+//       we effectively treat it as a no-op for now. When hardware devices that natively support
+//       mid-circuit measurements in an arbitrary basis are available, we will create a new
+//       QuantumDevice to implement this functionality according to the hardware specs.
+RESULT *__catalyst__mbqc__measure_in_basis(QUBIT *wire, uint32_t plane, double angle,
```

I will look at it more closely next week, but just very quickly, do we just want no-op, or explicit failure, or do we want to silently fall back onto the standard computational basis measurement? The latter could be a bit confusing for users, especially considering the different semantics.

---

My goal here was to "simulate" the execution pipeline as much as possible (and not to simulate the actual physics). My thinking was that a computational-basis measurement operation will give us a better sense of cpu/memory usage than a complete no-op. Also, this isn't a user-facing feature at the moment, it falls under the umbrella of experimental ftqc work.

---

Makes sense. In that case, should we at least give a warning here, saying "this function does not actually do what the name suggests"? Even this is not user-facing, we might want to make these potential points of confusion explicit.

---

Not sure if we have runtime warning infrastructure, but even just printing a message here would be fine with me.

---

Let's avoid warnings --- this isn't expected to be supported outside the team, and all failures resulting from incorrect use are fine :)

---

Sorry, posted my reply before refreshing the page and didn't see your comment! If we're happy **not** emitting warnings that a-okay with me :+1:

---

I see, all good then!